cdap-data-fabric/src/main/java/co/cask/cdap/data/stream/service/AbstractStreamService.java

```diff
@@ -97,4 +112,39 @@ protected ScheduledExecutorService executor() {
     executor = Executors.newSingleThreadScheduledExecutor(Threads.createDaemonThreadFactory("heartbeats-scheduler"));
     return executor;
   }
+
+  /**
+   * Get the size of events ingested by a stream since its creation, in bytes.
+   * @param streamId id of the stream
+   * @return Size of events ingested by a stream since its creation
+   * @throws IOException when getting an error retrieving the metric
+   */
+  protected long getStreamEventsSize(Id.Stream streamId) throws IOException {
+    MetricDataQuery metricDataQuery = new MetricDataQuery(
```

Comment:
The metric can ttl, but and we had the same problem before, stream data could ttl and we didn't really handle it well in the counting logic. So nothing has changed from that point of view. 

We don't handle truncation of the stream anymore, that's the point of this PR too. Notification subscribers have to be aware of that, and know that a notification is published whenever enough data has been ingested, no matter if that data has ttled already or if the stream has been truncated.